creator/ingest_runs/tasks/validation_run.py

```
@@ -1,82 +1,171 @@
+    ingest lib's extraction utility to clean the values and map the column
+    names to the standard set of concepts expected by the validator
+    """
+    logger.info(f"Performing extraction to clean and map file {version}")
 
-    Upload validation results and report files to storage backend
+    # Load ingest extract operations for this file
+    if not version.root_file:
+        raise Exception(
+            f"Validation failed. Version {version} missing root_file"
+        )
+    ec = version.extract_config_path
+    if not ec:
+        logger.warning(
+            f"⚠️  Could not find an extract config for file version {version} "
+            f"with file type: {version.root_file.file_type}. This file will "
+            "not be included in validation."
+        )
+        return None
+    # Extract file content into a DataFrame
+    try:
+        df = pandas.DataFrame(extract_data(version))
+    except Exception as e:
+        er_msg = (
+            f"Validation failed! Error in parsing {version} content into "
+            f"a DataFrame."
+        )
+        raise type(e)(er_msg).with_traceback(sys.exc_info()[2])
+
+    logger.info(
+        f"Applied extract config {os.path.split(ec)[-1]} for file "
+        f"{version.root_file.file_type}, {version.kf_id}"
+    )
+
+    ext = Extractor()
+    ext.logger = logger
+
+    return ext.extract(df, ec)
+
+
+def validate_file_versions(validation_run):
+    """
+    Load ValidationRun.versions into DataFrames, extract data from each
+    version, clean and map the data before passing it to the validator
     """
-    logger.info(f"Begin validation for {validation_run.versions.all()}")
-    results = {}
+    versions = validation_run.versions.all()
+    logger.info(f"Begin validating file versions: {pformat(list(versions))}")
+    # Clean and map source data
+    df_dict = {}
+    for version in versions:
+        clean_df = clean_and_map(version)
```

No error handling?
Maybe it would be better to perform the skip logic here. If there's a problem trying to clean or map, throw a warning and skip it.

---

I would rather raise the Exceptions and log the warnings inside the `clean_and_map` method bc there are several different error cases and messages that have specific messages

---

I don't think you need to handle all exceptions here, just ones specific to determining whether to fail out or to skip a file.
This is the place where you're ultimately deciding which path to take, which is currently done with the `isinstance` check, but consider how something like this may be more readable and generalizable to other failure cases:
```python
for version in versions:
    try:
        df_dict[version.kf_id] = clean_df
    except ExtractConfigDoesNotExist:
        logger.warning("Extract configuration does not exist. Will not try to validate document")
        continue
    except Exception as err: # It would be better to have some sort of exception specific to extract_data here
        logger.error("Something went wrong trying to extract file contents")
        raise
```